scipy/io/arff/arffread.py

```diff
@@ -273,7 +273,7 @@ def _get_date_format(atrv):
             if "yyyy" in pattern:
                 pattern = pattern.replace("yyyy", "%Y")
                 datetime_unit = "Y"
-            elif "yy":
+            elif "yy" in pattern:
```

Comment:
For this change here, I should mention that there is both an open issue and an attempt to provide the same fix with a regression test:

https://github.com/scipy/scipy/issues/11864
https://github.com/scipy/scipy/pull/11888

Probably my fault for making it overly complicated and asking for a regression guard (which was proving much harder than the fairly-simple fix)

Reply:
I reverted this commit... Thanks!